enhancements/update/clusterversion-osus-signatures.md

```diff
@@ -0,0 +1,148 @@
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+CVO currently fetches the upstream urls for signature verification through [cluster-update-keys]. 
+To improve disconnected cluster user experience we need to provide a way for the CVO  to get the required signatures 
+from the OSUS running in the disconnected environment.
+
+
+## Motivation
+
+Today, in the disconnected environments, a user has to clone the signatures config map as well as the release images to update the clusters.
+With [serve image signatures for disconnected environments], OSUS will be able to serve the cloned image signatures locally
+
+Currently, there is no way for the CVO to fetch the signatures through custom URIs. This enhancement defines an API change in the CVO
+to accept a custom update-service URI which can be used to verify release signatures. 
+
+
+### User Stories
+
+* As a cluster administrator, I want the cluster's CVO to add a new signature store URI.
+* As a cluster administrator, I want the cluster's CVO to replace the existing signature store URIs with a user provided URI
+
+### Goals
+
+* To verify the release image signature in disconnected environments using OSUS signatures URI.
+* To replace existing signature stores with custom user-provided stores.
+
+### Non-Goals
+
+* This enhancement does not cover how OSUS running in the disconnected environment gets the signatures and if the 
+signatures are validated in the process
+
+## Proposal
+
+### Workflow Description
+
+1. Cluster admin will have to add the verification URI to CVO
+2. CVO will reach the newly added URI to verify the release images.
+
+### API Extensions
+
+* CVO will grow a new field, an array, which will be populated with the image verification URI.
+* The api extension will replace the existing stores. 
+* To use existing stores, admin will have to add them manually with locally hosted stores, although locally hosted stores 
+are not required for this functionality to work.
+* This will be an array as it'll allow the admin to use locally configured stores with the existing stores
+* Admins will still be able to use the existing config maps for signature verification. 
+
+### Risks and Mitigations
+
+* The responsibility of mirroring the right signatures is going to be same as mirroring the right payload as we expect 
+customers will use oc-mirror to mirror the release payload as well building the graph-data container image with the signatures.
+* Currently, customers use [oc-mirror] to mirror release payload and Cincinnati graph data information for OSUS running in 
+the disconnected environment.
+* [oc-mirror] already has the signature information when it mirrors the release payload. We will be reusing the same 
+image signatures to serve through the OSUS URI.
+
+### Drawbacks
+
+* The cluster admin will have to have their own due diligence to make sure that the release images and verification signatures are
```

Comment:
I don't understand what this is saying.  [Here][1] I cover the various layers of protection.  None of them seem like something where somebody managing an OpenShift Update Signature mirror, or admining a cluster that pulls signatures from that mirror, would need to take additional steps for safety.  There is the generic "if you don't mirror the signatures you need, then they won't be available for clusters to consume", but :shrug:, that's true of any signature-delivery pipeline, and not a drawback unique to this proposal.

[1]: https://github.com/openshift/enhancements/pull/1485#discussion_r1339269751

Reply:
I agree. I do not think  this enhancement introducing this drawback. We should remove this point.

Reply:
removed the drawbacks as this API extension won't be introducing them